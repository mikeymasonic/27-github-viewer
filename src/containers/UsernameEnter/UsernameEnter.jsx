import React, { Component } from 'react';
import UsernameInput from '../../components/UsernameInput/UsernameInput.jsx';

export default class UsernameEnter extends Component {

  state = {
    username: ''
  }

  handleUsernameChange = ({ target }) => {
    this.setState({ username: target.value });
  };

  handleUsernameSubmit = () => {

  };

  render() {
    return (
      <>
        <UsernameInput username={this.state.username} onUsernameChange={this.handleUsernameChange} onUsernameSubmit={this.handleUsernameSubmit} />
      </>
    );
  }
}
