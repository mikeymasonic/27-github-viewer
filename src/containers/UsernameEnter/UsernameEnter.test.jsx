import React from 'react';
import { shallow } from 'enzyme';
import UsernameEnter from './UsernameEnter';

describe('UsernameEnter component', () => {
  it('renders UsernameEnter', () => {
    const wrapper = shallow(<UsernameEnter 
      username="mikeymasonic"
      onUsernameChange={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
