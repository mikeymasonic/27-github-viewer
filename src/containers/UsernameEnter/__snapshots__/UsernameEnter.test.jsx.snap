// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsernameEnter component renders UsernameEnter 1`] = `
<Fragment>
  <UsernameInput
    onUsernameChange={[Function]}
    onUsernameSubmit={[Function]}
    username=""
  />
</Fragment>
`;
